glustolibs-gluster/glustolibs/gluster/samba_libs.py

```diff
@@ -435,6 +445,70 @@ def is_winbind_service_running(mnode):
     return False
 
 
+def setup_samba_ctdb_cluster(servers, primary_node,
+                             ctdb_volname,
+                             ctdb_nodes, ctdb_vips, ctdb_volume_config,
+                             all_servers_info):
+    """
+    Create ctdb-samba cluster if doesn't exists
+
+    Returns:
+        bool: True if successfully setup samba else false
+    """
+    # Check if ctdb setup is up and running
+    if is_ctdb_status_healthy(primary_node):
+        g.log.info("ctdb setup already up skipping "
+                   "ctdb setup creation")
+        return True
+    g.log.info("Proceeding with ctdb setup creation")
+    for mnode in servers:
+        ret = edit_hook_script(mnode, ctdb_volname)
+        if not ret:
+            return False
+        ret = enable_ctdb_cluster(mnode)
+        if not ret:
+            return False
+        ret = create_nodes_file(mnode, ctdb_nodes)
+        if not ret:
+            return False
+        ret = create_public_address_file(mnode, ctdb_vips)
+        if not ret:
+            return False
+    server_info = all_servers_info
```

Comment:
Can't we use the existing variables here? I see server_info and ctdb_config are used only on Line : 480. In that case, can these two statements be removed? 
Please correct me if I got it wrong.

Reply:
Makes sense to use existing variables. I will make the changes and submit the patch.